src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -2880,27 +2880,30 @@ public Void emitObjectType(
      * If a symbol refers to a "platform" type (e.g. Promise) emit it specially here and return
      * true.
      */
-    private boolean maybeEmitGlobalType(String name) {
+    private boolean maybeEmitGlobalType(String name, ObjectType type) {
       if (name == null) return false;
 
+      boolean foundGlobalType = false;
       if (PlatformSymbols.GLOBAL_SYMBOL_ALIASES.contains(name)) {
         // See comment on GLOBAL_SYMBOL_ALIASES.
         emit(getGlobalSymbolNamespacePrefix() + "Global" + name);
-        return true;
+        foundGlobalType = true;
+      } else if (PlatformSymbols.CLOSURE_TO_TYPESCRIPT.get(name) != null) {
+        emit(PlatformSymbols.CLOSURE_TO_TYPESCRIPT.get(name));
+        foundGlobalType = true;
+      } else if (PlatformSymbols.TYPESCRIPT_LIB_D_TS.contains(name)) {
+        // Assume it's safe to emit the name as-is.
+        emit(name);
+        foundGlobalType = true;
       }
 
-      String newName = PlatformSymbols.CLOSURE_TO_TYPESCRIPT.get(name);
-      if (newName != null) {
-        emit(newName);
-        return true;
+      if (foundGlobalType
```

Comment:
This is error-prone because, if the foundGlobalType wasn't checked first calling 'emit' would be very bad.

How about this function is renamed to 'maybeGlobalType' and keep it emit free. Move the eemitGenericTypeArguments to the two callers.

Reply:
Done.